### ISSUE TITLE
Blank lines created by o then esc are filled with trailing whitespace

### ISSUE THREAD

**Issue by mreishus** [VSCodeVim/Vim#849]
---

Please _thumbs-up_ 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

---
### What did you do?

While I was in the middle of some lines that are indented, I pressed:
o
ESC
j
Then saved. My intention is to make a new blank line and continue my editing elsewhere.  The file type is "Javascript React" but I suspect it happens with all file types.
### What did you expect to happen?

I expected the new line to be added to be completely blank.   VIM does this.
### What happened instead?

A line of nothing but spaces was added to my file.  I have to manually delete the spaces which is annoying.
### Technical details:
- VSCode Version: 1.5.3
- VsCodeVim Version: 0.3.0
- OS: Win 7

**Comment by FSMaxB** [comment]
Same when you are at the end of the line in insert mode and press <enter> twice. In vim, the first line gets all spaces stripped after pressing <enter> the second time.

**Comment by johnfn** [comment]
My Vim definitely does not do this (it does indent the file). Is there some settings that affect this?

**Comment by mreishus** [comment]
I tried using as minimal of a setup as I could think of by running

`vim -u NONE -N` (this loads no vimrc and then sets nocompat) and typing `:set smartindent` once loaded.

I went to an indented block in a ruby file and pressed `o` on a line.  My cursor was indented and I was ready to type in the right spot.  

![screenshot-1](http://i.imgur.com/kFxFqKy.png)

However, I pressed esc to leave insert mode without typing anything.

![screenshot-2](http://i.imgur.com/Gtcy5ug.png)

Now the cursor is back on the beginning of the line, and if I save the file, there will be no blank whitespace on this line.   

This is `VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jun 16 2016 10:50:38)`, and I remember it doing this for a long time without having to set any special setting.  I also just tested `:set autoindent` instead of smartindent and it also does the same behavior.
